zingolib/src/wallet/describe.rs

```diff
@@ -302,6 +304,55 @@ impl LightWallet {
             .map(|(_index, sk)| *sk)
             .collect::<Vec<_>>()
     }
+
+    #[allow(clippy::result_unit_err)]
+    /// gets a UnifiedAddress, the first the wallet. this is the only receiver implemented as 2024-09-22
+    pub fn get_first_ua(&self) -> Result<zcash_keys::address::UnifiedAddress, ()> {
+        if let Some(possible_ua) = self.wallet_capability().addresses().iter().next() {
+            Ok(possible_ua.clone())
+        } else {
+            Err(())
+        }
+    }
+
+    #[allow(clippy::result_unit_err)]
+    /// UnifiedAddress type is not a string. to process it into a string requires chain date.
+    pub fn encode_ua_as_pool(
+        &self,
+        ua: &zcash_keys::address::UnifiedAddress,
+        pool: PoolType,
+    ) -> Result<String, ()> {
+        match pool {
+            PoolType::Transparent => ua
+                .transparent()
+                .map(|taddr| {
+                    super::keys::address_from_pubkeyhash(&self.transaction_context.config, *taddr)
```

Comment:
non-blocking but this is deprecated, I used a different encoding path for the LRZ update. i can help with this in the future

Reply:
This conversation is outdated.